### ISSUE TITLE
请问：如何用deepseek模型微调或蒸馏自己的数据？还是用传统的lora? 官方好像没有介绍

### ISSUE THREAD

**Issue by alexhmyang** [deepseek-ai/DeepSeek-V3#623]

**Comment by kuazhangxiaoai** [comment]
同问

**Comment by lizhichao999** [comment]
import torch
import torch.nn as nn
import torch.optim as optim
import torchvision
import torchvision.transforms as transforms
 
# 设置设备
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
 
# 超参数
epochs = 20
batch_size = 256
temperature = 4  # 温度参数
alpha = 0.7      # 软标签损失权重
 
# 数据加载
transform = transforms.Compose([
    transforms.RandomCrop(32, padding=4),
    transforms.RandomHorizontalFlip(),
    transforms.ToTensor(),
    transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))
])
 
train_set = torchvision.datasets.CIFAR10(root='./data', train=True, download=True, transform=transform)
train_loader = torch.utils.data.DataLoader(train_set, batch_size=batch_size, shuffle=True)
 
# 定义老师模型（ResNet-18）
teacher = torchvision.models.resnet18(pretrained=True)
teacher.fc = nn.Linear(teacher.fc.in_features, 10)  # CIFAR-10有10类
teacher = teacher.to(device)
 
# 定义学生模型（MobileNetV2）
student = torchvision.models.mobilenet_v2(pretrained=True)
student.classifier[1] = nn.Linear(student.last_channel, 10)
student = student.to(device)
 
# 训练老师模型（此处假设老师已预训练好，直接加载）
# 实际中需要先训练老师模型，此处为简化跳过
 
# 定义损失函数和优化器
criterion_hard = nn.CrossEntropyLoss()           # 硬标签损失
criterion_soft = nn.KLDivLoss(reduction='batchmean')  # 软标签损失
optimizer = optim.Adam(student.parameters(), lr=0.001)
 
# 蒸馏训练循环
for epoch in range(epochs):
    teacher.eval()   # 固定老师模型
    student.train()  # 训练学生模型
    
    running_loss = 0.0
    for inputs, labels in train_loader:
        inputs, labels = inputs.to(device), labels.to(device)
        
        # 前向传播
        with torch.no_grad():
            teacher_logits = teacher(inputs)
        
        student_logits = student(inputs)
        
        # 计算损失
        # 软标签损失（使用温度参数软化）
        soft_loss = criterion_soft(
            nn.functional.log_softmax(student_logits / temperature, dim=1),
            nn.functional.softmax(teacher_logits / temperature, dim=1)
        ) * (alpha * temperature * temperature)  # 缩放损失
        
        # 硬标签损失
        hard_loss = criterion_hard(student_logits, labels) * (1 - alpha)
        
        total_loss = soft_loss + hard_loss
        
        # 反向传播
        optimizer.zero_grad()
        total_loss.backward()
        optimizer.step()
        
        running_loss += total_loss.item()
    
    print(f'Epoch {epoch+1}, Loss: {running_loss/len(train_loader):.4f}')
 
print("Distillation finished!") deepseek给的答案，请问如何通过deepseek的模型架构，知识蒸馏别的模型，如Qwen，有没有能运行的demo

**Comment by mowentian** [comment]
烦请移步其他社区以寻求支持
